gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/ProcessWorkunitsSource.java

```diff
@@ -0,0 +1,75 @@
+package org.apache.gobblin.temporal.ddm;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.common.eventbus.EventBus;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.gobblin.configuration.SourceState;
+import org.apache.gobblin.configuration.WorkUnitState;
+import org.apache.gobblin.instrumented.extractor.InstrumentedExtractor;
+import org.apache.gobblin.source.InfiniteSource;
+import org.apache.gobblin.source.extractor.DataRecordException;
+import org.apache.gobblin.source.extractor.Extractor;
+import org.apache.gobblin.source.workunit.WorkUnit;
+import org.apache.gobblin.stream.RecordEnvelope;
+
+
+@Slf4j
+public class ProcessWorkunitsSource implements InfiniteSource {
```

Comment:
`NoWorkunitsSource` or `EmptyWorkunitsSource`... doesn't seem to be anything processing specific (also sources don't ever "process" WUs themself, so may be confusing)

also javadoc (and note importance of `InfiniteSource`, which I presume a key element)